### ISSUE TITLE
Training error  Check failed: (temp.size()) >= (num_negative) in multibox_target.cc

### ISSUE THREAD

**Issue by OddEcust** [zhreshold/mxnet-ssd#28]
When I try to run train.py, the "Check failed: (temp.size()) >= (num_negative)" in multibox_target.cc happens. 
[11:43:24] D:\ODD\Code\MxNet\SSD_Master\mxnet\dmlc-core\include\dmlc/logging.h:235: [11:43:24] D:\ODD\Code\MxNet\SSD_Master\mxnet\src\operator\multibox_target.cc:217: Check failed: (temp.size()) >= (num_negative)
[11:43:24] D:\ODD\Code\MxNet\SSD_Master\mxnet\dmlc-core\include\dmlc/logging.h:235: [11:43:24] d:\odd\code\mxnet\ssd_master\mxnet\src\engine\./threaded_engine.h :306: [11:43:24] D:\ODD\Code\MxNet\SSD_Master\mxnet\src\operator\multibox_target.cc:217: Check failed: (temp.size()) >= (num_negative)

**Comment by PhanTask** [comment]
I got this error too. Waiting for help.

**Comment by PhanTask** [comment]
I tracked the error and found the problem in train/metric.py:

def update(self, labels, preds):
        """
        Implementation of updating metrics
        """
        # get generated multi label from network
        cls_prob = preds[0].asnumpy()
        loc_loss = preds[1].asnumpy()
        cls_label = preds[2].asnumpy()

When it goes to 'preds[0].asnumpy()' it failed. How to solve this?

Another thing I found is that I checked the 'preds':
[<NDArray 1x21x7308 @cpu(0)>, <NDArray 1x29232 @cpu(0)>, <NDArray 1x7308 @cpu(0)>]
But should it be 1x21x7308, 1x29232, 1x7308x4?(due to [ssd_vgg16_reduced.pdf](https://github.com/zhreshold/mxnet-ssd/blob/master/tools/ssd_vgg16_reduced.pdf) )

**Comment by zhreshold** [comment]
Are you trying to train on CPU? The multibox_target.cc is for prototyping and it is too slow that could take a year to train a model.

**Comment by zhreshold** [comment]
@PhanTask The shapes are correct, the last one is class label, not the anchors

**Comment by PhanTask** [comment]
@zhreshold Thank you, I got it.
And yes I'm using CPU on a server.Time to get a N card I think.

**Comment by OddEcust** [comment]
@zhreshold  @PhanTask  I'm using CPU too. Great thanks for you.

**Comment by aaronshan** [comment]
@PhanTask @OddEcust Do you resolved this problem? thanks for your reply.

**Comment by zhreshold** [comment]
CPU mode should be fixed now. But I doubt the usability of it.
